watertap/unit_models/translators/translator_asm2d_adm1.py

```diff
@@ -191,6 +190,15 @@ def build(self):
             mutable=True,
             doc="Smoothing factor",
         )
+        self.heaviside_k = Param(
+            initialize=1e1,
```

Comment:
Shall we provide better initial values for the users, maybe as the default? Or are the current initial values the preferred ones from other references?

Reply:
Ironically there was no improvement. This is likely because the flowsheet only operates under one of the conditionals in the expr_if, so it's almost like the other conditional doesn't even exist. So when we switched to this smooth function, there's no noticeable improvement. That's my rudimentary understanding, at least.

Reply:
What if we perturb the model? That is, you can write a test either for the translator itself or on BSM2-P where you vary inlet concentrations, for example (ideally ones that correspond to the components dealt with in expression ifs/smooth heaviside approximations) and see what the differences are

It adds on work, and we could otherwise just cross our fingers and hope that this works better as we move forward---but then we may have harder time later if things continue to go wrong. Would be good to determine what impact this limited modification has thus far before proceed--by doing a parameter sweep (or testing out that IPOPTConvergenceAnalysis capability?)

Reply:
Note-- i was responding to @Morgan88888888 's comment-- " I am wondering how much improvement was shown in solving the flowsheet model. Could you please comment on it?"

Reply:
@adam-a-a  Thanks, and I like your idea, and testing extreme cases is a good way to estimate potential improvement.

Reply:
I've added some IPOPTConvergenceAnalysis tests. The results are also identical to the version of the translator in main

Reply:
@adam-a-a @Morgan88888888 I've figured out why this PR doesn't seem to have any effect on the model diagnostics. It's because, in the original expr_if, a is ALWAYS larger than b. So it's as if the first conditional doesn't even exist. For b to be larger than A, we would need to change the parameter values themselves (specifically `Ni["X_pr"]` or `i_NXS`)

TLDR: These expr_ifs aren't causing any problems. If there were problematic expr_ifs, they've already been addressed in the previous PR where I replaced them with smooth_min or smooth_max.